src/test/functional/steps/list-contact-types.ts

```diff
@@ -0,0 +1,84 @@
+import {Given, Then, When} from 'cucumber';
+import * as I from '../utlis/puppeteer.util';
+import {expect} from 'chai';
+import {getFirstTableRowIndexContainingText} from '../utlis/puppeteer.util';
+
+When('I click on contact type list', async () => {
+  const selector = '#tab_contact-types';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.click(selector);
+});
+
+Given('I click edit contact type {string}', async (contactType: string) => {
+  const tableRow = await getFirstTableRowIndexContainingText('#contactTypeListContent', 1, contactType);
+  expect(tableRow).greaterThan(-1);
+  // The table row index returned is zero-based but nth-child works on a 1-based index so we add one.
+  const selector = `#contactTypeListContent > table > tbody > tr:nth-child(${tableRow + 1}) > td:nth-child(2) > a`;
+  expect(await I.checkElement(selector)).equal(true);
+  await I.click(selector);
+});
+
+Then('I am redirected to the contact type {string} form', async (editContactType: string) => {
+  const selector = '#contactTypeListContent > h2';
+  const formTitle = await I.getElementText(await I.getElement(selector));
+  expect(formTitle).equal(editContactType);
+});
+
+Given('I will make sure to clear entries for the Contact Type', async () => {
+  expect(await I.checkElement('#ct-type')).equal(true);
+  await I.clearField('#ct-type');
+
+  expect(await I.checkElement('#ct-type-cy')).equal(true);
+  await I.clearField('#ct-type-cy');
+});
+
+Then('I enter {string} in name textbox', async (name: string) => {
```

Comment:
You could have one function here that can be reused, might make future work easier too :) 

Maybe: 

```
(can pass in: "welsh" / '#ct-type-cy'  or "english" / "'#ct-type')
---
Then('I enter {string} in the element {string}', async (nameCy: string, element: string) => {
  expect(await I.checkElement(element)).equal(true);
  await I.setElementValueForInputField(element, nameCy);
});
```